File: CS/Basic/recursion/SquareSummation.go
package main

import (
	// "fmt"
	"math"
)

func squareSummation(n int32) int32 {
	if n <= 1 {
		return 1
	}

	return squareSummation(n-1) + int32(math.Pow(float64(n), 2))
}

// func main() {
// 	var n int32
// 	fmt.Scan(&n)
// 	fmt.Println(squareSummation(n))
// }

File: CS/Basic/recursion/SquareSummation.java
package CS.Basic.recursion;

public class SquareSummation {
    public static int squareSummation(int n) {
        if (n <= 1) {
            return 1;
        }

        return squareSummation(n - 1) + n * n;
    }

    public static void main(String[] args) {
        System.out.println(squareSummation(208));
    }
}
